apps/root/src/pages/strategy-guardian-detail/strategy-management/withdraw/market-withdraw-modal/index.tsx

```diff
@@ -1,9 +1,132 @@
 import useActiveWallet from '@hooks/useActiveWallet';
 import useContractService from '@hooks/useContractService';
-import { DisplayStrategy, EarnPermission, WithdrawType } from 'common-types';
-import React from 'react';
-import { FormattedMessage } from 'react-intl';
-import { Modal } from 'ui-library';
+import { AmountsOfToken, DisplayStrategy, EarnPermission, EarnPosition, Token, WithdrawType } from 'common-types';
+import { ContainerBox, Modal, Wallet2Icon, UnlockIcon, Typography, colors, Skeleton } from 'ui-library';
+import TokenIcon from '@common/components/token-icon';
+import { formatCurrencyAmount, formatUsdAmount } from '@common/utils/currency';
+import { useEarnManagementState } from '@state/earn-management/hooks';
+import { isUndefined } from 'lodash';
+import useEstimateMarketWithdraw from '../hooks/useEstimateMarketWithdraw';
+
+const WithdrawAmountItem = ({
+  title,
+  amount,
+  token,
+  intl,
+  isLoading,
+  isNegative,
+}: {
+  title: MessageDescriptor;
+  amount?: AmountsOfToken;
+  token?: Token;
+  intl: IntlShape;
+  isLoading?: boolean;
+  isNegative?: boolean;
+}) => (
+  <ContainerBox flexDirection="column" gap={1}>
+    <Typography variant="bodySmallRegular">{intl.formatMessage(title)}</Typography>
+    <ContainerBox gap={2} alignItems="center">
+      <TokenIcon token={token} size={6} />
+      <Typography variant="bodyBold" color={({ palette }) => colors[palette.mode].typography.typo2} lineHeight={1}>
+        {isLoading ? (
+          <Skeleton variant="text" sx={{ width: '10ch' }} />
+        ) : token && amount ? (
+          `${isNegative ? '-' : ''}${formatCurrencyAmount({
+            amount: amount.amount,
+            token,
+            intl,
+          })}`
+        ) : (
+          '-'
+        )}
+      </Typography>
+    </ContainerBox>
+    <Typography variant="labelLarge" color={({ palette }) => colors[palette.mode].typography.typo3}>
+      {isLoading ? (
+        <Skeleton variant="text" sx={{ width: '10ch' }} />
+      ) : !isUndefined(amount?.amountInUSD) ? (
+        `(${isNegative ? '-' : ''}$${formatUsdAmount({
+          amount: amount.amountInUSD,
+          intl,
+        })})`
+      ) : (
+        '-'
+      )}
+    </Typography>
+  </ContainerBox>
+);
+
+interface MarketWithdrawModalContentProps {
+  strategy?: DisplayStrategy;
+  isOpen: boolean;
+  position?: EarnPosition;
+}
+
+const MarketWithdrawModalContent = ({ strategy, isOpen, position }: MarketWithdrawModalContentProps) => {
+  const intl = useIntl();
+  const { asset } = useEarnManagementState();
+
+  const { isLoading, withdrawAmountOfToken, withdrawFeeAmountOfToken, estimatedMarketAmount } =
+    useEstimateMarketWithdraw({
+      strategy,
+      shouldRefetch: isOpen,
+      position,
+    });
+
+  return (
+    <ContainerBox flexDirection="column">
+      <ContainerBox gap={6}>
+        <CommonTransactionStepItem icon={<Wallet2Icon fontSize="large" />} isCurrentStep isLast={false} hideWalletLabel>
```

Comment:
does this actually work ok and adds the line in the left to all of this?

Reply:
Yep!
<img width="640" alt="Screenshot 2024-11-06 at 3 31 29 PM" src="https://github.com/user-attachments/assets/87ab4d34-35e9-44c7-8721-941dfc32680b">